lib/talktome.rb

```diff
@@ -26,7 +26,7 @@ def with_env(which, &bl)
   def set_env(which, value, &bl)
     old, ENV[which] = ENV[which], value
     bl.call.tap{
-      ENV[which] = old unless old.nil?
+      ENV[which] = old
     }
   end
   module_function :set_env
```

Comment:
I don't understand this code, if one day you have time to explain it to me

Reply:
Have a look at `Object#tap`, that's what's missing in your mental picture. Basically, the method takes a block, saves an env var before the block, executes and returns the block result, but restores the env var afterwards. Same as

```
old = x
x = y
result = bl.call
x = old
return result